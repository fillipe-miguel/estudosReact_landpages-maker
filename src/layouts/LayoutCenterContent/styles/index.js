import styled, { css } from 'styled-components';

export const LayoutCenterContentRoot = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const InnerHtml = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};
    text-align: justify;

    & h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      text-align: center;
    }
  `}
`;

export default LayoutCenterContentRoot;
