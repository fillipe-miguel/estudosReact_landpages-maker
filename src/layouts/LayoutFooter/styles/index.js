import styled, { css } from 'styled-components';

export const LayoutFooterRoot = styled.footer`
  ${({ theme }) => css`
    height: 8rem;
    text-align: center;
    font-size: ${theme.fonts.sizes.xsmall};
    background-color: ${theme.colors.white};

    a {
      pointer-events: all !important;
      text-decoration: underline !important;
    }
  `}
`;

export default LayoutFooterRoot;
