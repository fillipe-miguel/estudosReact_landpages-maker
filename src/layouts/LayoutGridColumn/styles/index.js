import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const LayoutGridColumnRoot = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    gap: ${theme.spacings.xxlarge};

    & ${HeadingStyles.HeadingRoot} {
      margin-bottom: ${theme.spacings.small};
    }
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      flex-direction: column;

      & ${HeadingStyles.HeadingRoot} {
        margin-bottom: ${theme.spacings.xlarge};
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    max-width: 35rem;
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      min-width: auto;
      text-align: center;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css`
    max-width: 64rem;
  `}
`;

export const Image = styled.img`
  ${() => css`
    max-width: 100%;
  `}
`;
