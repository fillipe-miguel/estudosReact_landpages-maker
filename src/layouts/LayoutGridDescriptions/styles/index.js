import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    & ${HeadingStyles.HeadingRoot} {
      position: relative;
      left: 5rem;
      margin: ${theme.spacings.medium} 0;
    }

    & ${HeadingStyles.HeadingRoot}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xxxlarge};
      left: -5rem;
      top: -2.5rem;
      transform: skew(-15deg);
    }
  `}
`;
