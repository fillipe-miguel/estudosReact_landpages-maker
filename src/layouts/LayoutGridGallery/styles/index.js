import styled, { css } from 'styled-components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.medium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
    height: fit-content;

    &:hover > img {
      transition: all 0.3s ease-in-out;
      transform: scale(1.2) rotate(9deg);
    }

    img {
      pointer-events: none;
      max-width: 100%;
    }
  `}
`;
