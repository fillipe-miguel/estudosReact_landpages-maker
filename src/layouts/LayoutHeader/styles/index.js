import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutHeaderRoot = styled.header`
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;

  & > ${SectionContainerStyles.SectionContainerRoot} {
    padding-top: ${({theme}) => theme.spacings.small};
    padding-bottom: ${({theme}) => theme.spacings.small};;
  }

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      bottom: 0;
      height: 100vh;
      visibility: ${props.visible ? 'visibility' : 'hidden'};
      opacity: ${props.visible ? '1' : '0'};
      transition: all 0.3s ease-in-out;

      & > ${SectionContainerStyles.SectionContainerRoot} {
        display: grid;
        align-items: center;

        height: 100%;
        margin: 0;

        overflow-y: auto;
      }

      & ${HeadingStyles.HeadingRoot} {
        padding-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      flex-direction: column;
      text-align: center;
      padding: ${theme.spacings.xxlarge};
    }
  `}
`;

export const ToggleMenuButton = styled.button`
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;

  display: none;

  font-size: 2.5rem;
  padding: 0.5rem;

  cursor: pointer;
  border: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      align-items: center;
    }
  `}
`;

export default LayoutHeaderRoot;
