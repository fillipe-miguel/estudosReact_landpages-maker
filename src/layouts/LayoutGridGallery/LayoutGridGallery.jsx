import P from 'prop-types';
import React from 'react';

import * as Styled from './styles';

import { Heading, TextComponent } from '../../components';
import { SectionBackground } from '../../sections';

function LayoutGridGallery({
  title,
  description,
  setBackground,
  grid,
  layoutId,
}) {
  return (
    <SectionBackground layoutId={layoutId} setBackground={setBackground}>
      <Styled.HeaderContainer>
        <Heading
          uppercase
          size="xxlarge"
          level="2"
          colorDarker={!setBackground}
        >
          {title}
        </Heading>
        <TextComponent asHtml colorDarker={!setBackground}>
          {description}
        </TextComponent>
      </Styled.HeaderContainer>
      <Styled.Grid>
        {grid.map((element) => (
          <Styled.GridElement key={element.srcImg}>
            <img src={element.srcImg} alt={element.altText} />
          </Styled.GridElement>
        ))}
      </Styled.Grid>
    </SectionBackground>
  );
}

LayoutGridGallery.defaultProps = {
  setBackground: false,
};

LayoutGridGallery.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  setBackground: P.bool,
  grid: P.arrayOf(
    P.shape({
      srcImg: P.string.isRequired,
      altText: P.string.isRequired,
    }),
  ).isRequired,
  layoutId: P.string.isRequired,
};

export default LayoutGridGallery;
