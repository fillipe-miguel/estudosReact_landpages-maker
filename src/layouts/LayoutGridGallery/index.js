import LayoutGridGallery from './LayoutGridGallery';

export default LayoutGridGallery;
