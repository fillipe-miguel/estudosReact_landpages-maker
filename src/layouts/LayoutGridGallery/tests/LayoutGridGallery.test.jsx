/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import LayoutGridGallery from '..';

import * as mocks from '../../../mocks';
import { MyThemeProvider, theme } from '../../../styles';

describe('<LayoutGridGallery/>', () => {
  // ================================= Teste 01 =================================
  it('Should render default', () => {
    render(
      <MyThemeProvider>
        <LayoutGridGallery {...mocks.layouts.gridGallery} />
      </MyThemeProvider>,
    );

    // Getting a random thing to comprove rendering
    expect(screen.queryAllByRole('img')).toHaveLength(6);
  });

  // ================================= Teste 02 =================================
  it('Should render with background set', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridGallery {...mocks.layouts.gridGallery} setBackground />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.primaryColor,
    });
  });

  // ================================= Snapshot =================================
  it('Should render with background set', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridGallery {...mocks.layouts.gridGallery} setBackground />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
