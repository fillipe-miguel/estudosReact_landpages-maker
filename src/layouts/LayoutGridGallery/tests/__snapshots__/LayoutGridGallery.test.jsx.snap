// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGridGallery/> Should render with background set 1`] = `
.c2 {
  margin-bottom: 4.8rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(250px,1fr));
  gap: 2.4rem;
}

.c6 {
  overflow: hidden;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c6:hover > img {
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transform: scale(1.2) rotate(9deg);
  -ms-transform: scale(1.2) rotate(9deg);
  transform: scale(1.2) rotate(9deg);
}

.c6 img {
  pointer-events: none;
  max-width: 100%;
}

.c3 {
  color: #FFF;
  text-transform: uppercase;
  font-size: 4.8rem;
}

.c4 {
  font-size: 1.6rem;
  color: #FFF;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
  background-color: #0A1128;
  color: #FFF;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        My grid gallery
      </h2>
      <p
        class="c4"
      >
        Aqui vai ter várias imagens e coisas legais ablublé ablubléablubléablubléablubléablubléablublé.....
      </p>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <img
          alt="imagem 1"
          src="https://source.unsplash.com/random/800x800?r=1"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="imagem 2"
          src="https://source.unsplash.com/random/800x800?r=2"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="imagem 3"
          src="https://source.unsplash.com/random/800x800?r=3"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="imagem 4"
          src="https://source.unsplash.com/random/800x800?r=4"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="imagem 5"
          src="https://source.unsplash.com/random/800x800?r=5"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="imagem 6"
          src="https://source.unsplash.com/random/800x800?r=6"
        />
      </div>
    </div>
  </div>
</div>
`;
